### PR TITLE
Exercise-3-7-code updated

### PR DESCRIPTION
**Pull request by michaelleoniuk**
The purpose of changes were to:

Use React Router to enable routing in your app:

- Install react-router and react-router-dom packages by running npm install react-router react-router-dom --save.

- Refactor the code in render() method in main-view.jsx to use a BrowserRouter, and add routes for your views.

- Replace buttons that use onMovieClick and onBackClick with React Router links. This includes MovieCard, and MovieView in main-view.jsx.

- Remove anything related to setSelectedMovie, including the state selectedMovie, since it will no longer be used.

Followed by:

Add a navigation bar at the top of the page.

- For unauthenticated users, provide links to the Login/Signup page

- For authenticated users, provide links to Home (your MainView), Profile and Logout.

Add a user Profile view (linked to the main navigation menu for authenticated users). As per your project brief, this view should:

- Display user information. You can obtain the logged in user information by using the /users endpoint and then filtering the returning list by comparing the usernames;

- Allow a user to update their user information (username, password, email, date of birth);

- Allow a user to deregister;

- Display a user's favorite movies as a list. For this you’ll need to filter the movies array with something like let favoriteMovies = movies.filter(m => user.FavoriteMovies.includes(m._id)). With this, favoriteMovies is just a regular array of movie objects which could, for example, be displayed as movie cards (the MovieCard components would need to be passed to the ProfileView component and rendered there);

- Add a “Favorite” button to your MovieCard and/or MovieView components, so that logged in users can select a movie to store in their list of favorites;

- Allow a user to remove a movie from their list of favorites.

### CODE REVIEW

File: src/components/main-view/main-view.jsx [hunk 1]
```diff
@@ -3,99 +3,229 @@ import { MovieCard } from "../movie-card/movie-card";
...
+            },
+          };
+        });
+        setMovies(moviesFromApi);
+      });
+  }, [token]);
+
+  const addFav = (id) => {
+    fetch(
+      `https://czo-myflix-ccfb67c11465.herokuapp.com/${user.Username}/movies/${id}`,
```
Comment (Stanley-Okwii) [inline src/components/main-view/main-view.jsx]: Kindly update this...
```suggestion
      `https://czo-myflix-ccfb67c11465.herokuapp.com/users/${user.Username}/movies/${id}`,
```

File: src/components/main-view/main-view.jsx [hunk 2]
```diff
@@ -3,99 +3,229 @@ import { MovieCard } from "../movie-card/movie-card";
...
-                            Name: movie.Director.Name
-                        }
-                    };
-                });
-                setMovies(moviesFromApi);
-            });
-    }, [token]);
+  const removeFav = (id) => {
+    fetch(
+      `https://czo-myflix-ccfb67c11465.herokuapp.com/${user.Username}/movies/${id}`,
```
Comment (Stanley-Okwii) [inline src/components/main-view/main-view.jsx]: Even here...
```suggestion
      `https://czo-myflix-ccfb67c11465.herokuapp.com/user/${user.Username}/movies/${id}`,
```

File: src/components/movie-card/movie-card.jsx [hunk 3]
```diff
@@ -1,25 +1,40 @@
...
+        Title: PropTypes.string,
+        Director: PropTypes.shape({
+            Name: PropTypes.string,
+        }),
+        _id: PropTypes.string,
     }).isRequired,
-    onMovieClick: PropTypes.func.isRequired
+    addFav: PropTypes.func.isRequired,
+    removeFav: PropTypes.func.isRequired,
+    isFavorite: PropTypes.bool.isRequired,
```
Comment (Stanley-Okwii) [inline src/components/movie-card/movie-card.jsx:39]: Thanks for adding these here

File: src/components/main-view/main-view.jsx (hunk 1 above)
Comment (michaelleoniuk) [reply src/components/main-view/main-view.jsx]: Done

File: src/components/main-view/main-view.jsx (hunk 2 above)
Comment (michaelleoniuk) [reply src/components/main-view/main-view.jsx]: Done

File: src/components/movie-card/movie-card.jsx (hunk 3 above)
Comment (michaelleoniuk) [reply src/components/movie-card/movie-card.jsx:39]: Done

**Pull request merged**